Assets/Samples/InputRecorder/InputRecorderInspector.cs

```diff
@@ -0,0 +1,234 @@
+                var newDefaultSizeInMB = EditorGUILayout.IntSlider(m_DefaultSizeText, defaultSizeInMB, 1, 100);
+                if (newDefaultSizeInMB != defaultSizeInMB)
+                    m_CaptureMemoryDefaultSizeProperty.intValue = newDefaultSizeInMB * 1024 * 1024;
+
+                var maxSizeInMB = m_CaptureMemoryMaxSizeProperty.intValue / (1024 * 1024);
+                var newMaxSizeInMB = maxSizeInMB < newDefaultSizeInMB
+                    ? newDefaultSizeInMB
+                    : EditorGUILayout.IntSlider(m_MaxSizeText, maxSizeInMB, 1, 100);
+                if (newMaxSizeInMB != maxSizeInMB)
+                    m_CaptureMemoryMaxSizeProperty.intValue = newMaxSizeInMB * 1024 * 1024;
+
+                EditorGUILayout.PropertyField(m_RecordButtonPath, m_RecordButtonText);
+                EditorGUILayout.PropertyField(m_PlayButtonPathProperty, m_PlayButtonText);
+
+                if (scope.changed)
+                    serializedObject.ApplyModifiedProperties();
+            }
+
+            EditorGUILayout.Space();
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                ////TODO: go-to-next and go-to-previous button
+                // Play and pause buttons.
+                EditorGUI.BeginDisabledGroup(recorder.eventCount == 0 || recorder.captureIsRunning);
+                var oldIsPlaying = recorder.replayIsRunning;
+                var newIsPlaying = GUILayout.Toggle(oldIsPlaying, !oldIsPlaying ? m_PlayText : m_StopText, EditorStyles.miniButton,
+                    GUILayout.Width(50));
+                if (oldIsPlaying != newIsPlaying)
+                {
+                    if (newIsPlaying)
+                        recorder.StartReplay();
+                    else
+                        recorder.StopReplay();
+                }
+                if (newIsPlaying && recorder.replay != null && GUILayout.Button(recorder.replay.paused ? m_ResumeText : m_PauseText, EditorStyles.miniButton,
+                    GUILayout.Width(50)))
+                {
+                    if (recorder.replay.paused)
+                        recorder.StartReplay();
+                    else
+                        recorder.PauseReplay();
+                }
+                EditorGUI.EndDisabledGroup();
+
+                // Record button.
+                EditorGUI.BeginDisabledGroup(recorder.replayIsRunning);
+                var oldIsRecording = recorder.captureIsRunning;
+                var newIsRecording = GUILayout.Toggle(oldIsRecording, m_RecordText, EditorStyles.miniButton, GUILayout.Width(50));
+                if (oldIsRecording != newIsRecording)
+                {
+                    if (newIsRecording)
+                        recorder.StartCapture();
+                    else
+                        recorder.StopCapture();
+                }
+                EditorGUI.EndDisabledGroup();
+
+                ////TODO: finish
```

Comment:
Fix TODO here.

Reply:
Fix how? Ultimately I'd like to be able to inspect the contents of any trace with something better than that simple tree view in the input debugger and I started on that here but then decided it makes more sense to defer this I disabled the code that was already there.

Reply:
I'm fine with //TODO comments in code. But in my experience, ifdefed out or commented out code never ages well and by the time you get to revisit it you have to rewrite it anyways. So I would either do it now, or remove the code, but not ship disabled code like this.

Reply:
Removed.